### PR TITLE
validation on Select field

### LINKED ISSUE #8204: Add validation on Select field option keys
## Scope & Context

We can input special characters in Select API values 

+ We need to change the label "API keys" to "API values" as it is misleading

image

### PR DESCRIPTION
**Pull request by ketanMehtaa**
fix #8204
I changed "API keys" to "API values".
Stopped inputting special characters in Select field option keys.

@lucasbordeau please check the changes and tell me if I need to do any other changes. :)

### CODE REVIEW

File: packages/twenty-front/src/modules/settings/data-model/fields/forms/select/utils/getOptionValueFromLabel.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export const getOptionValueFromLabel = (label: string) => {
   try {
     transliteratedLabel = computeOptionValueFromLabelOrThrow(label);
   } catch (error) {
-    return label;
+    return "";
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/settings/data-model/fields/forms/select/utils/getOptionValueFromLabel.ts:10]: What are you trying to do here? You said you'd tackle "Stopped inputting special characters in Select field option keys" but that's not really what you're doing here?

[the remark this answers is not in the document]
Comment (FelixMalfait) [reply packages/twenty-front/src/modules/settings/data-model/fields/forms/select/utils/getOptionValueFromLabel.ts:10]: My bad, I sent a comment previously saying it wouldn't work. It seems it does but I don't understand why haha. Will check the code more in depth

File: packages/twenty-front/src/modules/settings/data-model/fields/forms/select/utils/getOptionValueFromLabel.ts [hunk 2]
```diff
@@ -1,14 +1,13 @@
-import snakeCase from 'lodash.snakecase';
-
 import { computeOptionValueFromLabelOrThrow } from '~/pages/settings/data-model/utils/compute-option-value-from-label.utils';
 
 export const getOptionValueFromLabel = (label: string) => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/data-model/fields/forms/select/utils/getOptionValueFromLabel.ts:3]: remove this helper

File: packages/twenty-front/src/pages/settings/data-model/utils/compute-metadata-name-from-label.utils.ts [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
-  return transliterateAndFormatOrThrow(label, METADATA_NAME_VALID_PATTERN);
+
+  const formattedString = slugify(prefixedLabel, {
+    trim: true,
+    separator: '_',
+    allowedChars: 'a-zA-Z0-9',
+  });
+
+  if (formattedString === '') {
+    throw new Error('Invalid label');
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/data-model/utils/compute-metadata-name-from-label.utils.ts:18]: add the label value

File: packages/twenty-front/src/pages/settings/data-model/utils/compute-option-value-from-label.utils.ts [hunk 4]
```diff
@@ -1,6 +1,17 @@
-import { OPTION_VALUE_VALID_PATTERN } from '~/pages/settings/data-model/constants/OptionValueValidPattern';
-import { transliterateAndFormatOrThrow } from '~/pages/settings/data-model/utils/transliterate-and-format.utils';
+import { slugify } from 'transliteration';
 
 export const computeOptionValueFromLabelOrThrow = (label: string): string => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/data-model/utils/compute-option-value-from-label.utils.ts:3]: fileName is not following the function name

Comment (charlesBochet) [reply packages/twenty-front/src/pages/settings/data-model/utils/compute-option-value-from-label.utils.ts:3]: remove orThrow actually

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/utils/validate-object-metadata-input.util.ts [hunk 5]
```diff
@@ -134,9 +109,33 @@ const validateNameCharactersOrThrow = (name?: string) => {
 };
 
 export const computeMetadataNameFromLabelOrThrow = (label: string): string => {
-  const formattedString = transliterateAndFormatOrThrow(label);
+  if (!label) {
+    throw new ObjectMetadataException(
+      'Name is required',
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/metadata-modules/object-metadata/utils/validate-object-metadata-input.util.ts:114]: Label is required

**Review by charlesBochet — APPROVED** [review]

**Review by FelixMalfait — APPROVED** [review]
Thank you! Ended up cleaning a bit more than anticipated :)

**Pull request merged**